src/libs/actions/IOU.ts

```diff
@@ -112,7 +112,33 @@ type TrackExpenseInformation = {
     actionableWhisperReportActionIDParam?: string;
     onyxData: OnyxData;
 };
-
+type CategorizeTrackedExpenseInformation = {
```

Comment:
@dukenv0307 I investigated those PR and the type definitions introduced in those PR.

1. In theory We could reuse RequestMoneyTransactionParams instead of new type definition CategorizeTrackedExpenseTransactionParams, Coz all the extra fields in RequestMoneyTransactionParams compared to CategorizeTrackedExpenseTransactionParams are optional. But those extra fields are not any how handled by categorizeTrackedExpense() function and in future it could be misunderstood.

2. Other Types like RequestMoneyPolicyParams, RequestMoneyParticipantParams, etc... has no good overlap with input params for categorizeTrackedExpense() function.

3. From investigating other PRs, I think what is expected here is to pass only one object to the function and pass few clusters as sub-objects.
For the same we could rename CategorizeTrackedExpenseInformation to CategorizeTrackedExpenseParams and include transactionParams and createdWorkspaceParams.
Something like...
```js
type CategorizeTrackedExpenseParams = {
    policyID: string;
    transactionID: string;
    moneyRequestPreviewReportActionID: string;
    moneyRequestReportID: string;
    moneyRequestCreatedReportActionID: string;
    actionableWhisperReportActionID: string;
    linkedTrackedExpenseReportAction: OnyxTypes.ReportAction;
    linkedTrackedExpenseReportID: string;
    transactionThreadReportID: string;
    reportPreviewReportActionID: string;
    onyxData: OnyxData | undefined;
    isDraftPolicy: boolean;
    transactionParams: CategorizeTrackedExpenseTransactionParams,
    createdWorkspaceParams?: CreateWorkspaceParams,
};
.
.
.
function categorizeTrackedExpense(
    trackedExpenseParams: CategorizeTrackedExpenseParams
) {
.
.
.
```

Reply:
I agree.

Reply:
We could also group following ids as an object and name it trackedExpenseReportInformation 
```js
    moneyRequestPreviewReportActionID: string;
    moneyRequestReportID: string;
    moneyRequestCreatedReportActionID: string;
    actionableWhisperReportActionID: string;
    linkedTrackedExpenseReportAction: OnyxTypes.ReportAction;
    linkedTrackedExpenseReportID: string;
    transactionThreadReportID: string;
    reportPreviewReportActionID: string;
```

Reply:
After those changes code should look like..
```js
type CategorizeTrackedExpenseTransactionParams = {
    transactionID: string;
    amount: number;
    currency: string;
    comment: string;
    merchant: string;
    created: string;
    taxCode: string;
    taxAmount: number;
    category?: string;
    tag?: string;
    billable?: boolean;
    receipt?: Receipt;
};
type CategorizeTrackedExpensePolicyParams = {
    policyID: string;
    isDraftPolicy: boolean;
};
type CategorizeTrackedExpenseReportInformation = {
    moneyRequestPreviewReportActionID: string;
    moneyRequestReportID: string;
    moneyRequestCreatedReportActionID: string;
    actionableWhisperReportActionID: string;
    linkedTrackedExpenseReportAction: OnyxTypes.ReportAction;
    linkedTrackedExpenseReportID: string;
    transactionThreadReportID: string;
    reportPreviewReportActionID: string;
};
type CategorizeTrackedExpenseParams = {
    onyxData: OnyxData | undefined,
    trackedExpenseReportInformation:CategorizeTrackedExpenseReportInformation,
    transactionParams: CategorizeTrackedExpenseTransactionParams,
    policyParams: CategorizeTrackedExpensePolicyParams,
    createdWorkspaceParams?: CreateWorkspaceParams,
};
.
.
.
function categorizeTrackedExpense(
    trackedExpenseParams: CategorizeTrackedExpenseParams
) {
.
.
```

Reply:
Changes are up.